cms/src/main/java/uk/nhs/digital/apispecs/swagger/CodegenDefaultGenerator.java

```diff
@@ -0,0 +1,194 @@
+package uk.nhs.digital.apispecs.swagger;
+
+import static freemarker.template.utility.Collections12.singletonList;
+import static java.util.stream.Collectors.toList;
+
+import io.swagger.codegen.v3.CodegenConstants;
+import io.swagger.codegen.v3.CodegenOperation;
+import io.swagger.codegen.v3.DefaultGenerator;
+import io.swagger.v3.oas.models.Components;
+import io.swagger.v3.oas.models.Operation;
+import io.swagger.v3.oas.models.PathItem;
+import io.swagger.v3.oas.models.Paths;
+import io.swagger.v3.oas.models.media.Schema;
+import io.swagger.v3.oas.models.parameters.Parameter;
+import io.swagger.v3.oas.models.security.SecurityRequirement;
+import io.swagger.v3.oas.models.security.SecurityScheme;
+import io.swagger.v3.oas.models.tags.Tag;
+
+import java.io.File;
+import java.util.*;
+
+/**
+ * Some producer teams require ordering their endpoints in a non-lexographical order.
+ * This requires a custom extension of the DefaultGenerator provided by CodeGen.
+ *
+ * This class CodegenDefaultGenerator extends DefaultGenerator class and sorts CodegenOperations before returning them.
+ * See the method orderedByDeclaredTags.
+ */
+public class CodegenDefaultGenerator extends DefaultGenerator {
+
+    @Override
+    public Map<String, List<CodegenOperation>> processPaths(Paths paths) {
```

Comment:
How about replacing this method (and the chain of duplicated private ones with something like this?:
```
    @Override
    public Map<String, List<CodegenOperation>> processPaths(Paths paths) {
        return orderedByDeclaredTags(super.processPaths(paths));
    }
```

Reply:
oh that's much better! i've slightly rewritten that just to avoid a bunch of nested method calls
```
@Override
public Map<String, List<CodegenOperation>> processPaths(Paths paths) {
    Map<String, List<CodegenOperation>> ops = super.processPaths(paths);
    return orderedByDeclaredTags(ops);
}
```
i say "a bunch" - i realise this only avoids one whole level of nesting :sweat_smile: